### PR TITLE
Edit inflow outflow and add dg

### LINKED ISSUE #21: Refine the Developer Guide with v1.0 changes
Update the Developer Guide with the introduction of v1.0 features.

### LINKED ISSUE #63: [TA Code Review] for LEONG XING YU
- [ ] Try to define methods first before using them.
- [ ] Abstract the methods a bit more if they are doing very similar things (e.g. addition or subtraction of item quantity)

### PR DESCRIPTION
**Pull request by leongxingyu**
Refactor inflow and outflow command classes
Add UML diagrams for both classes

### CODE REVIEW

File: docs/diagrams/OutflowCommandSequence.puml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+activate stockpal STOCKPAL_COLOR
+
+user -[USER_COLOR]> ui : "outflow 2 a/10"
+activate ui UI_COLOR
+
+ui -[UI_COLOR]>stockpal : "outflow 2 a/10"
+
+stockpal -[STOCKPAL_COLOR]> parser : parseCommand("outflow 2 a/10")
+activate parser PARSER_COLOR
+
```
Comment (Kobot7) [inline docs/diagrams/OutflowCommandSequence.puml:21]: Perhaps show the construction of OutflowCommand from parser?

File: docs/diagrams/OutflowCommandSequence.puml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+@startuml
+!include style.puml
+skinparam ArrowFontStyle plain
+
+Actor User as user USER_COLOR
+Participant ":UI" as ui UI_COLOR
+Participant ":StockPal" as stockpal STOCKPAL_COLOR
+Participant ":Parser" as parser PARSER_COLOR
+Participant ":Commands" as commands COMMANDS_COLOR
```
Comment (Kobot7) [inline docs/diagrams/OutflowCommandSequence.puml:9]: There is no class called Commands, did you mean Command or OutflowCommand?

File: docs/diagrams/OutflowCommandSequence.puml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+deactivate data
+
+data -[DATA_COLOR]> data : updateDecreaseQuantity(quantityClass, decreaseBy)
+activate data DATA_COLOR
+data --[DATA_COLOR]> data
+deactivate data
+
+data --[DATA_COLOR]> commands
+
+commands --[COMMANDS_COLOR]> ui
```
Comment (Kobot7) [inline docs/diagrams/OutflowCommandSequence.puml]: Command calls Ui, not returns to Ui

File: docs/diagrams/OutflowCommandSequence.puml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
```
Comment (Kobot7) [inline docs/diagrams/OutflowCommandSequence.puml:47]: Command needs to return to StockPal

**Review by EdmundTangg — COMMENTED** [review]
LGTM.

**Pull request merged**